### PR TITLE
add task type controller infrastructure

### PR DESCRIPTION
**Pull request by Glunky**

### CODE REVIEW

File: src/DashboardService.Business/TaskType/Interfaces/IEditTaskTypeCommand.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using LT.DigitalOffice.DashboardService.Models.Dto.Requests.TaskType;
+using LT.DigitalOffice.Kernel.Responses;
+using Microsoft.AspNetCore.JsonPatch;
+using System;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.TaskType.Interfaces;
+
+public interface IEditTaskTypeCommand
```
Comment (VLiamin) [inline src/DashboardService.Business/TaskType/Interfaces/IEditTaskTypeCommand.cs:11]: AutoInjection?

File: src/DashboardService/Controllers/TaskTypesController.cs [hunk 2]
```diff
@@ -37,6 +38,7 @@ public class TaskTypesController : ControllerBase
   [HttpPatch("{id}")]
   public async Task<OperationResultResponse<bool>> Patch(
     [FromRoute] Guid id, 
+    [FromQuery] GetTaskTypesFilter filter,
```
Comment (VLiamin) [inline src/DashboardService/Controllers/TaskTypesController.cs]: Get filter in patch request?

Comment (Glunky) [reply src/DashboardService/Controllers/TaskTypesController.cs]: fixed))

File: src/DashboardService.Business/TaskType/Interfaces/IEditTaskTypeCommand.cs (hunk 1 above)
Comment (AnnPostnikova) [reply src/DashboardService.Business/TaskType/Interfaces/IEditTaskTypeCommand.cs:11]: and not only here

File: src/DashboardService.Models.Dto/Requests/TaskType/CreateTaskTypeRequest.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+namespace LT.DigitalOffice.DashboardService.Models.Dto.Requests.TaskType;
+
+public record CreateTaskTypeRequest
+{
+  public string Name { get; set; }
```
Comment (AnnPostnikova) [inline src/DashboardService.Models.Dto/Requests/TaskType/CreateTaskTypeRequest.cs:8]: you can add an attribute [Required]

File: src/DashboardService.Business/TaskType/Interfaces/IEditTaskTypeCommand.cs (hunk 1 above)
Comment (Glunky) [reply src/DashboardService.Business/TaskType/Interfaces/IEditTaskTypeCommand.cs:11]: ++

File: src/DashboardService.Models.Dto/Requests/TaskType/CreateTaskTypeRequest.cs (hunk 3 above)
Comment (Glunky) [reply src/DashboardService.Models.Dto/Requests/TaskType/CreateTaskTypeRequest.cs:8]: ++

File: src/DashboardService.Business/TaskType/Interfaces/IGetTaskTypesCommand.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+using LT.DigitalOffice.Kernel.Responses;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.DashboardService.Business.TaskType.Interfaces;
+
+[AutoInject]
+public interface IGetTaskTypesCommand
+{
+  Task<FindResultResponse<TaskTypeInfo>> ExecuteAsync();
```
Comment (SolovyovDmitry) [inline src/DashboardService.Business/TaskType/Interfaces/IGetTaskTypesCommand.cs]: filter

**Review by SolovyovDmitry — APPROVED** [review]

File: src/DashboardService/Controllers/TaskTypesController.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  [HttpGet("{id}")]
+  public async Task<OperationResultResponse<TaskTypeInfo>> GetAsync(
+    [FromServices] GetTaskTypeCommand command,
+    [FromRoute] Guid id)
+  {
+    return await command.ExecuteAsync(id);
+  }
+
+  [HttpPatch("{id}")]
+  public async Task<OperationResultResponse<bool>> PatchAsync(
```
Comment (VLiamin) [inline src/DashboardService/Controllers/TaskTypesController.cs]: EditAsync?

**Review by VLiamin — APPROVED** [review]

**Pull request merged**